### DISCUSSION TITLE
Moving docs to a separate repo + using another tool for its maintaining

### DISCUSSION THREAD

**Discussion by andrii-bodnar** [lingui/js-lingui#1334] [Ideas]
I would like to discuss the possibility to move the Docs into a separate repo alongside its redesign and migrating to some modern documentation framework.

**Storing the documentation code along with the rest of the code (monolithic repository):**

_Pros_:
  - Easier to maintain and update documentation as it is in the same place as the code
  - Easier for contributors to make changes to both code and documentation at the same time

_Cons_:
  - Could make the repository larger and harder to navigate for those only interested in the code
  - May make it harder to separate concerns and track changes to documentation separately from code changes

**Storing the documentation code in a separate repository:**

_Pros_:
- Keeps the code repository clean and focused on the code
- Makes it easier to separate concerns and track changes to documentation separately from code changes
- Separate CI/CD processes (build, validation, deployment, etc)

_Cons_:
- May make it more difficult to keep documentation and code in sync
- Could make it more difficult for contributors to make changes to both code and documentation at the same time

---

Also, there are a lot of popular and great tools for docs management, for example:

- [docusaurus.io](https://docusaurus.io/)
- [nextra.site](https://nextra.site/)
- [vitepress.vuejs.org](https://vitepress.vuejs.org/)
- etc.

These tools are based on MD/MDX which is the leading way of managing documentation nowadays (this discussion has already had place before - #837).

In the case of using such tools, it makes even more sense to split the docs into a separate repository.

**Comment by fromthemills** [comment]
Think this is a very good change. I would love to help out. I could sponsor a version of the [Tailwind UI Protocol](https://tailwindui.com/templates/protocol) template. It's a more bespoke package than a docusaurus or nextra. But it keeps things simple and uses Next, MDX and tailwind as its core.

**Reply by andrii-bodnar** [reply]
@fromthemills thanks a lot for your suggestion but we decided to use Docusaurus and actually almost finished this - https://github.com/lingui/js-lingui/pull/1365

**Reply by fromthemills** [reply]
No probs! Looks great!

**Comment by andrii-bodnar** [comment]
https://lingui.dev 🚀
